container-storage-setup.sh

```diff
@@ -731,40 +729,47 @@ create_partition_parted(){
 }
 
 create_partition() {
-  local dev="$1"
+  local dev="$1" part
 
   if [ -x "/usr/sbin/parted" ]; then
     create_partition_parted $dev
   else
     create_partition_sfdisk $dev
   fi
 
+  part=$(query_first_child $dev)
+
   # Sometimes on slow storage it takes a while for partition device to
   # become available. Wait for device node to show up.
   if ! udevadm settle;then
     Fatal "udevadm settle after partition creation failed. Exiting."
   fi
 
-  if ! wait_for_dev ${dev}1; then
-    Fatal "Partition device ${dev}1 is not available"
+  if ! wait_for_dev ${part}; then
+    Fatal "Partition device ${part} is not available"
   fi
+  echo $part
+}
+
+query_first_child() {
+  lsblk -npl -o NAME "$1" | tail -n +2 | head -1
 }
 
 create_disk_partitions() {
-  local devs="$1"
+  local devs="$1" part
 
   for dev in $devs; do
-    create_partition $dev
+    part=$(create_partition $dev)
```

Comment:
We cannot drop support for `sfdisk`, that would break backward compatibility. Users who don't have `parted` will fall back to `sfdisk` for partitioning and we need to support that.